### PR TITLE
Follow button implementation for back end

### PR DESCRIPTION
**Pull request by kachalmers**
Added backend functionality for follow toggling. Can be tested with updates to front end https://github.com/kachalmers/software-engineering-react-s2-team1-project/pull/12

### CODE REVIEW

File: daos/FollowDao.ts [hunk 1]
```diff
@@ -66,12 +78,11 @@ export default class FollowDao implements FollowDaoI{
...
      */
-    findFollowByUsers = async (uid: string, ouid: string): Promise<Follow[]> =>
+    findFollowByUsers = async (uid: string, ouid: string): Promise<any> =>
         FollowModel
-            .find({follower: uid, followee: ouid})
+            .findOne({follower: uid, followee: ouid})
             .populate("follower")  // fill in information about follower
-            .populate("followee")   // fill in information about followee
-            .exec();
+            .populate("followee");   // fill in information about followee
```
Comment (mcartervt) [inline daos/FollowDao.ts:85]: Does this need a .exec()?

File: daos/Tuit2TagDao.ts [hunk 2]
```diff
@@ -102,7 +102,12 @@ export default class Tuit2TagDao implements Tuit2TagDaoI {
         if (tag !== null) {
             tuit2tags = await Tuit2TagModel
                 .find({tag: tag._id})
-                .populate("tuit")
+                .populate({
```
Comment (mcartervt) [inline daos/Tuit2TagDao.ts:105]: This is the same as in the other PR, right?

File: daos/FollowDao.ts [hunk 3]
```diff
@@ -47,6 +50,15 @@ export default class FollowDao implements FollowDaoI{
             .populate("followee")
             .exec();
 
+    findUsersFollowedByUser = async (uid: string): Promise<User[]> => {
```
Comment (mcartervt) [inline daos/FollowDao.ts:53]: Let's make a Controller method for this, just for Postman testing purposes.

**Review by mcartervt — CHANGES_REQUESTED** [review]
Tested userTogglesFollow() from FollowController via tests I added to Postman.  

Request that a Controller method be added for findUsersFollowedByUser() from the FollowDao, so it can be tested via Postman.

Otherwise it looks good!

**Comment by kachalmers** [comment]
> Tested userTogglesFollow() from FollowController via tests I added to Postman.
> 
> Request that a Controller method be added for findUsersFollowedByUser() from the FollowDao, so it can be tested via Postman.
> 
> Otherwise it looks good!

New commit to resolve this issue: https://github.com/kachalmers/software-engineering-node-s2-team1-project/commit/ca713df495755970e123e099de692c564619aa7a

Postman test for this controller method created in follows folder.

**Review by mcartervt — APPROVED** [review]
Changes are all good!

**Review by klalmeida — APPROVED** [review]
This looks great Katie!!

**Pull request merged**